### PR TITLE
Use theforeman-rubocop gemspec

### PR DESCRIPTION
**Pull request by archanaserver**
Inherited strictest.yml here, this rules have NewCops enable config. Also dropped Style/HashSyntax, Gemspec/RequireMFA and Style/SymbolProc from here because that was either included in the inherit file or was not needed.

This is part of Rubocop standerdization, link for the reference:
https://community.theforeman.org/t/standardizing-rubocop-with-theforeman-rubocop/37239
https://github.com/theforeman/theforeman-rubocop/?tab=readme-ov-file#all-opinionated-cops-with-new-ones---strictest

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,26 +1,18 @@
...
-  NewCops: enable
+  TargetRubyVersion: 2.7
 
 Layout/LineLength:
   Enabled: false
 
 Metrics:
   Enabled: false
 
-Style/Documentation:
```
Comment (stejskalleos) [inline .rubocop.yml:13]: Can we remove all other exceptions and have the config file the same as in the https://github.com/theforeman/hammer-cli-foreman-google/blob/master/.rubocop.yml?

Comment (archanaserver) [reply .rubocop.yml:13]: @stejskalleos in that case, i'll have to resolve the offenses created by cops like Layout/LineLength, Style/StringLiterals, and Style/FrozenStringLiteralComment. do you want it to be resolved?

Comment (stejskalleos) [reply .rubocop.yml:13]: Yop, that would be great

Comment (archanaserver) [reply .rubocop.yml:13]: okay, i'll add that changes in this PR, as well as others

**Comment by archanaserver** [comment]
@stejskalleos i've updated it with some new changes.

**Review by stejskalleos — APPROVED** [review]

**Comment by stejskalleos** [comment]
Thanks @archanaserver

**Pull request merged**